docs/contribute/ml-handlers.mdx

```diff
@@ -6,58 +6,54 @@ sidebarTitle: Building an ML Engine
 
 <Note>
-    "Database Handlers"
+    **Database Handlers**
+
     To learn more about handlers and how to implement a database handler, visit our [doc page here](/contribute/integrations/).
 </Note>
 
 ## Creating a Machine Learning Handler
 
-You can create your own ML handler within MindsDB by inheriting from [the `BaseMLEngine` class](https://github.com/mindsdb/mindsdb/blob/3d9090acb0b8b3b0e2a96e2c93dad436f5ebef90/mindsdb/integrations/libs/base.py#L123).
+You can create your own ML handler within MindsDB by inheriting from the [BaseMLEngine](https://github.com/mindsdb/mindsdb/blob/3d9090acb0b8b3b0e2a96e2c93dad436f5ebef90/mindsdb/integrations/libs/base.py#L123) class.
 
-By providing implementation for some or all of the methods contained in the `BaseMLEngine` class, you can connect with the machine learning library or framework of your choice.
+By providing the implementation for some or all of the methods contained in the `BaseMLEngine` class, you can connect with the machine learning library or framework of your choice.
 
 ### Core Methods
 
-Apart from the `__init__()` method, there are five methods, of which two must be implemented. Let's review the purpose of each method.
+Apart from the `__init__()` method, there are five methods, of which two must be implemented. We recommend checking actual examples in the codebase to get an idea of what goes into each of these methods, as they can change a bit depending on the nature of the system being integrated.
+
+Let's review the purpose of each method.
 
 | Method            | Purpose                                                                              |
 |-------------------|--------------------------------------------------------------------------------------|
 | `create()`        | It creates a model inside the engine registry.                                       |
 | `predict()`       | It calls a model and returns prediction data.                                        |
 | `update()`        | Optional. It updates an existing model without resetting its internal structure.     |
-| `describe()`      | Optional. It provides global models insights.                                        |
+| `describe()`      | Optional. It provides global model insights.                                         |
 | `create_engine()` | Optional. It connects with external sources, such as REST API.                       |
 
-### Implementing Mandatory Methods
+Authors can opt for adding private methods, new files and folders, or any combination of these to structure all the necessary work that will enable the core methods to work as intended.
+
+<Tip>
+    **Other Common Methods**
+    
+    Under the `mindsdb.integrations.libs.utils` library, contributors can find various methods that may be useful while implementing new handlers.
+</Tip>
```

Comment:
Similarly to the general intro page, here we should add a tip that mentions there is a wrapper for `BaseMLEngine`s called `BaseMLEngineExec` which is automatically deployed and takes care of modifying the data responses here into something that can be used alongside data handlers.